### ISSUE TITLE
Cufflinks is frozen at 14%

### ISSUE THREAD

**Issue by RxLoutre** [cole-trapnell-lab/cufflinks#72]
Hi everyone,

This is my first time using Cufflinks.

I'm trying to get a fasta corresponding to the RNA-seq from different tissues from D. suzukii. I first used hisat2 in order to align my reads against a reference, here the command I used in Hisat2 : 

`hisat2 --dta-cufflinks -p 25 --max-intronlen 50000 -x /media/DATAPART1/Documents/suzukii_assembly/annotation/indexes/suz_canu3 -1 '/media/DATAPART1/Documents/suzukii_assembly/annotation/EST_evidences/suz_ant.R1.fastq.gz','/media/DATAPART1/Documents/suzukii_assembly/annotation/EST_evidences/suz_ovi.R1.fastq.gz','/media/DATAPART1/Documents/suzukii_assembly/annotation/EST_evidences/suz_pro.R1.fastq.gz','/media/DATAPART1/Documents/suzukii_assembly/annotation/EST_evidences/suz_tar.R1.fastq.gz' -2 '/media/DATAPART1/Documents/suzukii_assembly/annotation/EST_evidences/suz_ant.R2.fastq.gz','/media/DATAPART1/Documents/suzukii_assembly/annotation/EST_evidences/suz_ovi.R2.fastq.gz','/media/DATAPART1/Documents/suzukii_assembly/annotation/EST_evidences/suz_pro.R2.fastq.gz','/media/DATAPART1/Documents/suzukii_assembly/annotation/EST_evidences/suz_tar.R2.fastq.gz' | samtools view -Sbo suzukii_rnaseq_ibdm.bam -`

Then I sorted the bam file using this command : 
`
samtools sort /media/DATAPART1/Documents/suzukii_assembly/annotation/suzukii_rnaseq_ibdm.bam /media/DATAPART1/Documents/suzukii_assembly/annotation/suzukii_rnaseq_ibdm_sorted.bam`

And then it seems to work for a moment, but it froze at 14% of the sequence (I tried several times, and it ran all over the night) : 

```
You are using Cufflinks v2.2.1, which is the most recent release.
[17:08:27] Inspecting reads and determining fragment length distribution.
Processing Locus tig00000383:187849-193055   [***                      ]  14%
```

Cufflinks didn't stop, it didn't throw any error, it is still doing stuff apparently (I'm watching it with htop), but it run all night, and I think that something went wrong...

So, is there something wrong with this particular locus ? Is the sorting step a problem ? Or I've just used the wrong library type ?

Thanks for your help !

Cheers,

Roxane

**Comment by aechchiki** [comment]
Hi @RxLoutre . I am doing basically the same thing as you, but with data from D. melanogaster, and I experienced the same issue as you: check [here](https://groups.google.com/forum/#!topic/tuxedo-tools-users/eU-sAUWOHEQ). I got no answer but found a way out.

Things I would suggest are: 
- improve your alignment command, aka: 
  - run Hisat2 three times separately, each time specifying ONE single R1/R2 couple (it was suggested to me by ), or
  - merge all the R1 to one big file and all the R2 to another file
- use [PicardTools](https://broadinstitute.github.io/picard/) for sorting your file in a Cufflinks-friendly way (check my recent issue on this tracker: https://github.com/cole-trapnell-lab/cufflinks/issues/71). using samtools is not wrong, but in my experience was about 3x slower
- depending on your coverage, Cufflinks will have very different runtime, which appeared to be quite normal and not to worry about, as suggested by experienced [users](http://seqanswers.com/forums/member.php?u=46235)

So basically, don't worry and let Cufflinks do its job. I could spot CPU usage but no tmp files for some days (I have 251929648 PE (x2) that survived after trimming, and Cufflinks has been running for 10 days on server).

Cheers.

**Comment by RxLoutre** [comment]
I tried something shorter with tophat2 : 

tophat2 --num-threads 30 --no-discordant --max-intron-length 50000 -o /media/loutre/SUZUKII/annotation/tophat_out/pro /media/loutre/SUZUKII/annotation/indexes/cleaned_canu3_suzukii '/media/loutre/SUZUKII/annotation/evidences/suz_pro.R1.fastq.gz' '/media/loutre/SUZUKII/annotation/evidences/suz_pro.R2.fastq.gz'

I added the --no-discrodant option because I thought it could improve Cufflinks treatments, but even with a subset of my RNAseq data and this option, cufflinks is still frozen at the same contig (tig0000382:15043-15157) and I'm really sure that it should'nt took that much time...

A collegue of mine already align our RNAseq data against an other assembly, and it took only one night, with a lower computer... So it shouldn’t take that long on my computer which have 32 cores, I think.

Maybe something is wrong with my assembly, how can I know ? 

Does anyone have an idea ? I don't think that letting cufflinks runs during 10 days is a good idea.
